gateway-ha/src/main/java/io/trino/gateway/ha/router/QueryCountBasedRouter.java

```diff
@@ -224,7 +224,11 @@ private synchronized Optional<String> getBackendForRoutingGroup(String routingGr
     @Override
     public String provideAdhocBackend(String user)
     {
-        return getBackendForRoutingGroup("adhoc", user).orElseThrow();
+        return getBackendForRoutingGroup("adhoc", user).orElseThrow(() -> {
+            RouterException adhocRoutingGroupException = new RouterException("did not find any cluster for the adhoc routing group");
+            log.error(adhocRoutingGroupException, adhocRoutingGroupException.getMessage());
+            return adhocRoutingGroupException;
```

Comment:
Logging an error and throwing an exception is weird. I would recommend removing logging.

Reply:
sure, I was logging and throwing at the same time in case other systems rely on the format of the log